optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/chained/TailChainSwapMove.java

```diff
@@ -38,20 +38,124 @@
 public class TailChainSwapMove<Solution_> extends AbstractMove<Solution_> {
 
     protected final GenuineVariableDescriptor<Solution_> variableDescriptor;
-    protected final SingletonInverseVariableSupply inverseVariableSupply;
-    protected final AnchorVariableSupply anchorVariableSupply;
 
     protected final Object leftEntity;
+    protected final Object leftValue;
+    protected final Object leftAnchor;
+
+    protected final Object rightEntity; // Sometimes null
     protected final Object rightValue;
+    protected final Object rightAnchor;
+
+    protected final boolean sameAnchor;
+    protected final Object leftNextEntity;
+    protected final Object rightNextEntity;
+    protected final boolean reverseAnchorSide;
+    protected final Object lastEntityInChain;
+    protected final Object entityAfterAnchor;
 
     public TailChainSwapMove(GenuineVariableDescriptor<Solution_> variableDescriptor,
```

Comment:
The constructor now includes a non-trivial logic, would it be possible to include a unit test for it? The current tests focus on existing constructor with the supply.

Reply:
This is the constructor with the supply's - that's the one heavily used by the existing unit test.
Do you mean the other constructors?

Reply:
Sorry, I confused the constructors. Truth, this one is tested in unit tests, the new constructors are not. And there is not much to test in them - maybe in connection with `rebase()`.